src/main/java/com/refinedmods/refinedstorage/apiimpl/network/grid/CraftingGridBehavior.java

```diff
@@ -71,28 +82,47 @@ public void onCrafted(INetworkAwareGrid grid, ICraftingRecipe recipe, PlayerEnti
 
     @Override
     public void onCraftedShift(INetworkAwareGrid grid, PlayerEntity player) {
+        CraftingInventory matrix = grid.getCraftingMatrix();
+        INetwork network = grid.getNetwork();
         List<ItemStack> craftedItemsList = new ArrayList<>();
-        int amountCrafted = 0;
         ItemStack crafted = grid.getCraftingResult().getStackInSlot(0);
 
         int maxCrafted = crafted.getMaxStackSize();
 
-        ForgeHooks.setCraftingPlayer(player);
+        int amountCrafted = 0;
+        boolean useNetwork = network != null;
+
+        IStackList<ItemStack> networkCraftingItems = null;
+        if (useNetwork) {
+            // We need a modifiable list of the items in storage that are relevant for this craft.
+            networkCraftingItems = createFilteredItemList(network,matrix);
```

Comment:
Maybe add a comment stating that we create a filtered list for performance reasons.

Reply:
Formatting `network,matrix`